Adds compat checker for versioning and compatibility

ibndawood opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Closes #2639.

This PR uses the `compat-checker` library from the [WooCommerce Grow repo](https://github.com/woocommerce/grow) to run compatibility and versioning checks during the plugin initialisation.

### Checks:
<!-- Mark completed items with an [x] -->
* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Screenshots:

Screenshots are for reference. **WooCommerce Brands** will be replaced by **Facebook for WooCommerce**.

#### WooCommerce not installed
![image](https://github.com/woocommerce/woocommerce-brands/assets/179533/524375c3-19da-4ffa-9867-5586f0a6d931)

#### WooCommerce not activated

![image](https://github.com/woocommerce/woocommerce-brands/assets/179533/d5fb4239-c0d5-445a-9c4b-c7bf68ffa003)

#### WooCommerce not compatible

![image](https://github.com/woocommerce/woocommerce-brands/assets/179533/0fb93163-536b-4b22-aa10-f340b1817ee6)

#### WooCommerce not tested

![image](https://github.com/woocommerce/woocommerce-brands/assets/179533/bc7eff36-82b2-4e8f-b4da-a2992460f366)

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

#### Pre-requisites

The PR requires deleting transients. A plugin like [Transients Manager](https://wordpress.org/plugins/transients-manager/) is recommended to delete the transients between tests.

#### WooCommerce activation tests

With the above code snippets in place, run the following tests. The instructions are for Facebook for WooCommerce but can be used for other WooGrow Extensions.

1. Install and activate Facebook for WooCommerce without installing WooCommerce.
2. Deactivate WooCommerce.
3. Verify the notice `Facebook for WooCommerce requires WooCommerce to be installed and activated. Please install WooCommerce.` is displayed.
4. Install WooCommerce without activating.
5. Verify the notice `Facebook for WooCommerce requires WooCommerce to be activated. Please activate WooCommerce.` is displayed.

#### WooCommerce versioning tests
⚠️  **Important note**: You will need to erase the transient `_transient_wc_grow_compat_checker_[EXTENSION].php[VERSION` after each change!!

1. Install and activate WooCommerce.
2. Install and activate Facebook for WooCommerce.
3. Modify the `WC requires at least` header field and set it to a version higher than the active WooCommerce version. Let's say the `WC requires at least` is set to `8.2` and the active WooCommerce version is `8.0.2`.
4. Verify the notice `Facebook for WooCommerce requires WooCommerce version 8.2 or higher. Please update WooCommerce to the latest version, or download the minimum required version` is displayed.
5. Modify the `WC tested up to` and set it to a lower version than the active WooCommerce version. Let's say the `WC tested up to` is set to `7.8`, the active WooCommerce version is `8.0.2` and Facebook for WooCommerce version is `3.0.34`.
6. Verify the warning notice `Facebook for WooCommerce - 3.0.34 is untested with WooCommerce 8.0.2` is displayed.

#### L2 support recommendation notice

1. Download the latest minus three WooCommerce version and activate it. Let's say `7.6`.
2. Verify the warning notice `Heads up! Facebook for WooCommerce will soon discontinue support for WooCommerce 7.6. Please update WooCommerce to take advantage of the latest updates and features.`

#### WordPress versioning tests
⚠️  **Important note**: You will need to erase the transient `_transient_wc_grow_compat_checker_[EXTENSION].php[VERSION` after each change!!

1. Modify the `Tested up to` and set it to a lower version than the current WordPress version. Let's say the `Tested up to` is set to `6.2`, the active WordPress version is `6.3`, and the Facebook for WooCommerce version is `3.0.34`.
2. Verify the warning notice `Facebook for WooCommerce - 3.0.34 is untested with WordPress 6.3` is displayed.

### Additional Details

To delete a transient using Transients Manager, navigate to Tools > Transients, search for `wc_grow` and delete the transient.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Add - Versioning and compatibility checks to implement support policy.

rawdreeg commented on class-wc-facebookcommerce.php:
```diff
@@ -9,8 +9,7 @@
 
-require_once __DIR__ . '/vendor/autoload.php';
-require_once __DIR__ . '/includes/fbutils.php';
+ require_once __DIR__ . '/includes/fbutils.php';
```
```suggestion
require_once __DIR__ . '/includes/fbutils.php';
```

rawdreeg commented on facebook-for-woocommerce.php:
```diff
@@ -127,7 +129,7 @@ public function __wakeup() {
 	public function init_plugin() {
 
-		if ( ! $this->plugins_compatible() ) {
+		if ( ! Checker::instance()->is_compatible( __FILE__, self::PLUGIN_VERSION ) || ! $this->plugins_compatible() ) {
```
Shouldn't Checker::instance()->is_compatible( __FILE__, self::PLUGIN_VERSION ) move inside plugins_compatible ?

ibndawood replied:
This is a good catch, @rawdreeg. My approach was not to disturb existing checks. However, in this case, we can remove the `plugins_compatible` method and replace it with `Checker::instance()->is_compatible( FILE, self::PLUGIN_VERSION )` alone.

ibndawood replied:
I've implemented the above suggestion in this commit: https://github.com/woocommerce/facebook-for-woocommerce/pull/2640/commits/8eefaf340d5c5f310805b7b0dc07f9c40e3e46fc

ibndawood commented:
Thank you, @rawdreeg, for taking the time to review the PR. I've pushed a couple of commits addressing your comments. Please let me know your thoughts.

rawdreeg commented on facebook-for-woocommerce.php:
```diff
@@ -127,7 +129,7 @@ public function __wakeup() {
 	public function init_plugin() {
 
-		if ( ! $this->plugins_compatible() ) {
+		if ( ! Checker::instance()->is_compatible( __FILE__, self::PLUGIN_VERSION ) ) {
```
This probably means we can remove plugins_compatible as this is a private method and it's no longer used :-)

rawdreeg reviewed (approved):
Looks good, Thanks @ibndawood . I added a suggestion to remove the plugins_compatible method. but that's left up to you :-)
